src/infrasys/time_series_models.py

```diff
@@ -377,8 +614,116 @@ def get_time_series_type_str() -> str:
         return "SingleTimeSeriesScalingFactor"
 
 
+class DeterministicMetadata(TimeSeriesMetadata):
+    """Defines the metadata for DeterministicTimeSeries and DeterministicSingleTimeSeries."""
+
+    initial_timestamp: datetime
+    resolution: timedelta
+    interval: timedelta
+    horizon: timedelta
+    window_count: int
+    type: Literal["DeterministicTimeSeries", "DeterministicSingleTimeSeries"]
+
+    @staticmethod
+    def get_time_series_data_type() -> Type[TimeSeriesData]:
+        """Return the data type associated with this metadata type."""
+        return DeterministicTimeSeries
+
+    @staticmethod
+    def get_time_series_type_str() -> str:
+        """Return the time series type as a string."""
+        return "DeterministicTimeSeries"
+
+    @classmethod
+    def from_data(
+        cls, time_series: DeterministicTimeSeriesType, **features: Any
+    ) -> "DeterministicMetadata":
+        """Construct a DeterministicMetadata from a DeterministicTimeSeries."""
+        units = (
+            QuantityMetadata(
+                module=type(time_series.data).__module__,
+                quantity_type=type(time_series.data),
+                units=str(time_series.data.units),
+            )
+            if isinstance(time_series.data, pint.Quantity)
+            else None
+        )
+
+        # I am not sure if this is the right way to do this.
+        type_str: Literal["DeterministicSingleTimeSeries", "DeterministicTimeSeries"] = (
```

Comment:
Is the `Literal` needed?

Reply:
It just to say that those are the only two valid. is it better if I remove it?